### DISCUSSION TITLE
Problem with `MaximizeExpressionOnFinalState` in `HierarchicalProblem` with the Aries backend

### DISCUSSION THREAD

**Discussion by nicola-gigante** [aiplan4eu/unified-planning#615] [Q&A] [answered]
Following the answer by @arbimo on #614 I've tried to use the `MaximizeExpressionOnFinalState` metric and I'm having troubles making it work.

This is a minimal example:

```
from unified_planning.shortcuts import *
import unified_planning.model.htn as htn
import unified_planning.engines as engines
import unified_planning.model.metrics as metrics

problem = htn.HierarchicalProblem()
problem.epsilon = 1

f = Fluent('f', IntType())
problem.add_fluent(f, default_initial_value = 0)

a = InstantaneousAction('a')
a.add_increase_effect(f, 1)

nop = InstantaneousAction('nop')

t = htn.Task('t')
m = htn.Method('m')
m.set_task(t)
m.add_subtask(a)

m_nop = htn.Method('m-nop')
m_nop.set_task(t)

problem.add_action(a)
problem.add_action(nop)
problem.add_task(t)
problem.add_method(m)
problem.add_method(m_nop)

problem.add_quality_metric(metrics.MaximizeExpressionOnFinalState(f))

problem.task_network.add_subtask(t)

with AnytimePlanner(
    name='aries', anytime_guarantee="INCREASING_QUALITY"
) as planner:
    for solution in planner.get_solutions(problem):
        print(solution)
```

When I run it I get this output:

```
status: INTERMEDIATE
engine: aries
plan: Hierarchical TimeTriggeredPlan:
status: SOLVED_OPTIMALLY
engine: aries
plan: Hierarchical TimeTriggeredPlan:
```

That is, the engine says the optimal plan is empty. This means the `m_nop` method is used to perform the task `t`. But the `m` method would have executed action `a` which has an increasing effect on `f`, which is the objective to maximize.

So I don't understand what's happening here. What am I doing wrong?

Thanks,
Nicola

**Comment by arbimo** [comment, marked as answer]
Thanks for the report, this is definitely a bug on Aries side, fixed here : https://github.com/plaans/aries/pull/139
In a nutshell, maximization was still interpreted as minimization.

The fix should be available in the development version in a few minutes.

**Reply by nicola-gigante** [reply]
Thanks!
